core/src/main/java/jenkins/security/ConfidentialStore.java

```diff
@@ -51,6 +55,9 @@
      */
     protected abstract @CheckForNull byte[] load(ConfidentialKey key) throws IOException;
 
+    // TODO consider promoting to public, and offering a default implementation of randomBytes (via the usual Util.isOverridden binary compat trick)
+    abstract SecureRandom secureRandom();
```

Comment:
Being a public API, this would sort of break it, though I have no idea if anyone besides me has even tried extending this class besides the default one.

Reply:
> this would sort of break it

Not if you used the usual `Util.isOverridden` trick. Both methods concrete, one unconditionally delegates to the other, vice-versa conditionally, else throw `AbstractMethodError` or supply a default implementation.

Reply:
Oh, that's a new trick to me.